File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.java
import java.util.*;

class Solution {
    public int solution(int[] A, int[] B) {
        int answer = 0;
        Arrays.sort(A);
        Arrays.sort(B);
        
        int bIndex = B.length - 1;

        for(int i=A.length-1;i>=0;i--) { //뒤에서부터 비교 -> B가 큰 경우에만 인덱스를 줄여주고 answer++
            if(B[bIndex]>A[i]) {
                answer++;
                bIndex--;
            }
        }
        return answer;
    }
}

File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.js
function solution(A, B) {

    let answer = 0;

    A.sort((a, b) => a - b);
    B.sort((a, b) => a - b);

    let bIndex = B.length - 1;

    // 뒤에서부터 비교
    for (let i = A.length - 1; i >= 0; i--) {

        // B가 더 큰 경우에만 승리
        if (bIndex >= 0 && B[bIndex] > A[i]) {
            answer++;
            bIndex--;
        }
    }

    return answer;
}